internal/defact/default.go

```diff
@@ -0,0 +1,262 @@
+		return target
+	}
+
+	oldExt := filepath.Ext(target)
+	return target[0:len(target)-len(oldExt)] + ".bat"
+}
+
+func createShimFile(fpath, shimPath string) error {
+	logging.Debug("Shimming %s at %s", fpath, shimPath)
+	exe, err := os.Executable()
+	if err != nil {
+		return errs.Wrap(err, "Could not get State Tool executable")
+	}
+
+	tplParams := map[string]interface{}{
+		"exe":     exe,
+		"command": filepath.Base(fpath),
+		"target":  fpath,
+	}
+	box := packr.NewBox("../../assets/shim")
+	boxFile := "shim.sh"
+	if rt.GOOS == "windows" {
+		boxFile = "shim.bat"
+	}
+	shimBytes := box.Bytes(boxFile)
+	shimStr, err := strutils.ParseTemplate(string(shimBytes), tplParams)
+	if err != nil {
+		return errs.Wrap(err, "Could not parse %s template", boxFile)
+	}
+
+	err = ioutil.WriteFile(shimPath, []byte(shimStr), 0755)
+	if err != nil {
+		return errs.Wrap(err, "failed to write shim command %s", shimPath)
+	}
+	return nil
+}
+
+// createShimFiles creates shims in the target path of all executables found in the bins dir
+// It overwrites existing files, if the overwrite flag is set.
+// On Windows the same executable name can have several file extensions,
+// therefore executables are only shimmed if it has not been shimmed for a
+// target (with the same or a different extension) from a different directory.
+// Also: Only the executable with the highest priority according to pathExt is shimmed.
+func createShimFiles(targetPath string, exePaths []string) error {
+	for _, exePath := range exePaths {
+		shimPath := shimTargetPath(targetPath, exePath)
+
+		// The link should not exist as we are always rolling back old shims before we run this code.
+		if fileutils.TargetExists(shimPath) {
+			logging.Error("Cannot create shim as target already exists: %s.", shimPath)
+			continue
+		}
+
+		if err := createShimFile(exePath, shimPath); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// SetupDefaultActivation sets symlinks in the global bin directory to the currently activated runtime
+func SetupDefaultActivation(cfg DefaultConfigurer, output output.Outputer, envGetter runtime.EnvGetter) error {
+	env, err := envGetter.GetEnv(false, "")
+	if err != nil {
+		return err
+	}
+
+	// roll back old symlinks
+	rollbackShims(cfg)
+
+	// Retrieve artifact binary directory
+	var bins []string
+	if p, ok := env["PATH"]; ok {
+		bins = strings.Split(p, string(os.PathListSeparator))
+	}
+
+	exes, err := Executables(bins)
+	if err != nil {
+		return locale.WrapError(err, "err_symlink_exes", "Could not detect executables")
+	}
+
+	// Remove duplicate executables as per PATH and PATHEXT
+	exes, err = UniqueExes(exes, os.Getenv("PATHEXT"))
+	if err != nil {
+		return locale.WrapError(err, "err_unique_exes", "Could not detect unique executables, make sure your PATH and PATHEXT environment variables are properly configured.")
+	}
+
+	binDir := config.GlobalBinPath()
+
+	output.Print(locale.Tl("default_shim", "Writing default installation to {{.V0}}.", binDir))
```

Comment:
This should be in the runner logic, not here. I don't think we need to pass output into this package at all.